specification/billing/resource-manager/Microsoft.Billing/stable/2024-04-01/agreement.json

```diff
@@ -0,0 +1,321 @@
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-fA-F0-9]{1,12}$"
+          },
+          {
+            "$ref": "./types.json#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A billing account.",
+            "schema": {
+              "$ref": "#/definitions/Agreement"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "AgreementByName": {
+            "$ref": "./examples/agreementByName.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/agreements": {
+      "get": {
+        "tags": [
+          "Agreement"
+        ],
+        "description": "Lists the agreements for a billing account.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/billing/"
+        },
+        "operationId": "Agreements_ListByBillingAccount",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "./types.json#/parameters/billingAccountName"
+          },
+          {
+            "$ref": "./types.json#/parameters/apiVersionParameter"
+          },
+          {
+            "in": "query",
+            "name": "expand",
+            "description": "May be used to expand the participants.",
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Result of listing agreements.",
+            "schema": {
+              "$ref": "#/definitions/AgreementListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "AgreementsListByBillingAccount": {
+            "$ref": "./examples/agreementsListByBillingAccount.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Agreement": {
+      "description": "An agreement.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./types.json#/definitions/ProxyResourceWithTags"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "The properties of a(n) Agreement",
+          "allOf": [
+            {
+              "$ref": "#/definitions/AgreementProperties"
+            }
+          ],
```

Comment:
Please replace all the following `"allOff"` with the relevant properties `"ref"`
```suggestion
         "$ref": "#/definitions/AgreementProperties"
```

![image](https://github.com/Azure/azure-rest-api-specs/assets/46921893/6d6493a6-7948-4eed-9449-eb12c82a7d65)

Reply:
Updated the PR